osdep/macosx_touchbar.m

```diff
@@ -0,0 +1,202 @@
+        tbItem.slider.maxValue = 100.0f;
+        tbItem.target = self;
+        tbItem.action = @selector(seekbarChanged:);
+        tbItem.customizationLabel = @"Seek Bar";
+        self.touchbarSeekbar = tbItem.slider;
+        return tbItem;
+    } else if ([identifier isEqualToString:play]) {
+        NSCustomTouchBarItem *tbItem = [self tbButtonWithId:identifier
+                                                       name:@"Play Button"
+                                                      image:NSImageNameTouchBarPauseTemplate
+                                                     action:@selector(play:)];
+        self.touchbarPlay = (NSButton *)tbItem.view;
+        return tbItem;
+    } else if ([identifier isEqualToString:previousItem]) {
+        return [self tbButtonWithId:identifier
+                               name:@"Previous"
+                              image:NSImageNameTouchBarGoBackTemplate
+                             action:@selector(previousItem:)];
+    } else if ([identifier isEqualToString:nextItem]) {
+        return [self tbButtonWithId:identifier
+                               name:@"Next"
+                              image:NSImageNameTouchBarGoForwardTemplate
+                             action:@selector(nextItem:)];
+    } else if ([identifier isEqualToString:previousChapter]) {
+        return [self tbButtonWithId:identifier
+                               name:@"Previous Chapter"
+                              image:NSImageNameTouchBarSkipBackTemplate
+                             action:@selector(previousChapter:)];
+    } else if ([identifier isEqualToString:nextChapter]) {
+        return [self tbButtonWithId:identifier
+                               name:@"Next Chapter"
+                              image:NSImageNameTouchBarSkipAheadTemplate
+                             action:@selector(nextChapter:)];
+    } else if ([identifier isEqualToString:cycleAudio]) {
+        return [self tbButtonWithId:identifier
+                               name:@"Cycle Audio"
+                              image:NSImageNameTouchBarAudioInputTemplate
+                             action:@selector(cycleAudio:)];
+    } else if ([identifier isEqualToString:cycleSubtitle]) {
+        return [self tbButtonWithId:identifier
+                               name:@"Cycle Subtitle"
+                              image:NSImageNameTouchBarComposeTemplate
+                             action:@selector(cycleSubtitle:)];
+    } else if ([identifier isEqualToString:currentPosition]) {
+        NSCustomTouchBarItem *tbItem = [self tbTextWithId:identifier name:@"Current Position"];
+        self.touchbarCurrentPosition = (NSTextField *)tbItem.view;
+        return tbItem;
+    } else if ([identifier isEqualToString:timeLeft]) {
+        NSCustomTouchBarItem *tbItem = [self tbTextWithId:identifier name:@"Time Left"];
+        self.touchbarTimeLeft = (NSTextField *)tbItem.view;
+        return tbItem;
+    }
+    return nil;
+}
```

Comment:
i am not aware of a different method. all the apple examples do it like this. it's probably because of the way the touchbar works, it being basically a small watchOS based device.